### PR TITLE
feat: 1310 Import Enforcement XML

### PR DESCRIPTION
**Pull request by BenAndrew-Kainos**

### CODE REVIEW

File: cases/enforcement_check/tests/test_export_xml.py [hunk 1]
```diff
@@ -56,6 +59,12 @@ def test_export_xml_with_parties_success(self):
             self.assertEqual(stakeholder["ORG_NAME"], party.organisation.name)
             self.assertEqual(stakeholder["PD_SURNAME"], party.name)
             self.assertEqual(stakeholder["ADDRESS1"], party.address)
+            # Ensure the correct EnforcementCheckID object is added for the import xml process
+            self.assertTrue(
+                EnforcementCheckID.objects.filter(
+                    id=stakeholder["SH_ID"], entity_id=entity_uuid, entity_type=party.type
+                ).exists()
+            )
```
Comment (efahy) [inline cases/enforcement_check/tests/test_export_xml.py:67]: Could this be made into a helper function?  Its repeated a few times

File: cases/enforcement_check/import_xml.py [hunk 2]
```diff
@@ -1,5 +1,103 @@
...
 
 
 def enforcement_id_to_uuid(id):
     return EnforcementCheckID.objects.get(id=id).entity_id
+
+
+def _extract_and_validate_xml_tree(tree):
+    data = []
+    try:
+        if tree.tag != BASE_TAG:
```
[the remark this answers is not in the document]
Comment (BenAndrew-Kainos) [reply cases/enforcement_check/import_xml.py]: Great suggestion. Done

File: Pipfile [hunk 3]
```diff
@@ -34,6 +34,7 @@ requests-toolbelt = "~=0.9"
 tblib = "~=1.6"
 weasyprint = "~=51.0"
 django-separatedvaluesfield = "~=0.6.0"
+defusedxml = "*"
```
[the remark this answers is not in the document]
Comment (BenAndrew-Kainos) [reply Pipfile]: Done

File: cases/enforcement_check/import_xml.py [hunk 4]
```diff
@@ -1,5 +1,103 @@
...
+from lite_content.lite_api.strings import Cases
+
+BASE_TAG = "SPIRE_UPLOAD"
+ENTITY_TAG = "SPIRE_RETURNS"
+APPLICATION_ID_TAG = "CODE1"
+ENTITY_ID_TAG = "CODE2"
+FLAG_TAG = "FLAG"
+
+
+def import_cases_xml(file):
```
[the remark this answers is not in the document]
Comment (BenAndrew-Kainos) [reply cases/enforcement_check/import_xml.py:23]: Done

File: cases/enforcement_check/import_xml.py [hunk 5]
```diff
@@ -1,5 +1,103 @@
...
+
+    return data
+
+
+def _convert_ids_to_uuids(data):
+    all_ids = set([item[APPLICATION_ID_TAG] for item in data] + [item[ENTITY_ID_TAG] for item in data])
+    uuids = EnforcementCheckID.objects.filter(id__in=all_ids).values("id", "entity_id", "entity_type")
+
+    if len(all_ids) != len(uuids):
+        raise ValidationError({"file": [Cases.EnforcementUnit.INVALID_ID_FORMAT]})
```
[the remark this answers is not in the document]
Comment (BenAndrew-Kainos) [reply cases/enforcement_check/import_xml.py]: Done

File: cases/enforcement_check/tests/test_export_xml.py (hunk 1 above)
Comment (BenAndrew-Kainos) [reply cases/enforcement_check/tests/test_export_xml.py]: Done

**Pull request merged**